backend/tessend/tessend.go

```diff
@@ -0,0 +1,829 @@
+		"url":           t.url,
+		"id":            data.Cluster.ID,
+		"metric_points": len(data.Metrics.Points),
+	}).Info("sending resource counts to tessen")
+
+	// send data
+	respHeader := t.send(data)
+	if respHeader == "" {
+		logger.Debug("no tessen response header")
+		return
+	}
+
+	// parse the response header for an integer value
+	interval, err := strconv.ParseUint(respHeader, 10, 32)
+	if err != nil {
+		logger.Debugf("invalid tessen response header: %v", err)
+		return
+	}
+
+	// validate the returned interval is within the upper/lower bound limits
+	err = corev2.ValidateInterval(uint32(interval))
+	if err != nil {
+		logger.Debugf("invalid tessen response header: %v", err)
+		return
+	}
+
+	// update the tessen interval if the response header returns a new value
+	if t.interval != uint32(interval) {
+		t.interval = uint32(interval)
+		logger.WithField("interval", t.interval).Debug("tessen interval updated")
+		t.interrupt <- t.config
+	}
+}
+
+// getDataPayload retrieves cluster, version, and license information
+// and returns the populated data payload.
+func (t *Tessend) getDataPayload() *Data {
+	// collect license information
+
+	// populate data payload
+	data := &Data{
+		Cluster: Cluster{
+			ID:           t.clusterID,
+			Distribution: Distribution,
+			Version:      version.Semver(),
+			License:      t.licenseGetter.Get(),
+		},
+	}
+
+	return data
+}
+
+// getPerResourceMetrics populates the data payload with the total number of each resource.
+func (t *Tessend) getPerResourceMetrics(now int64, data *Data) error {
+	var backendCount float64
+
+	// collect backend count
+
+	// TODO: This needs to be replaced with something else, that maybe we don't
+	// even have yet. Eric mentioned using the Operator Presence Control data,
+	// for which we don't really have an abstraction yet.
+
+	// cluster, err := t.client.Cluster.MemberList(t.ctx)
+	// if err != nil {
+	// 	logger.WithError(err).Error("unable to retrieve backend count")
+	// 	return err
+	// }
+	// if cluster != nil {
+	// 	backendCount = float64(len(cluster.Members))
+	// }
+	backendCount = 0
+
+	// populate data payload
+	mp := &corev2.MetricPoint{
+		Name:      "backend_count",
+		Value:     backendCount,
+		Timestamp: now,
+	}
+	appendInternalTag(mp)
+	appendStoreConfig(mp, t.GetStoreConfig())
+	logMetric(mp)
+	data.Metrics.Points = append(data.Metrics.Points, mp)
+
+	// loop through the entity class counts
+	data.Metrics.Points = append(data.Metrics.Points, t.getEntityClassMetrics(now)...)
+
+	// loop through the resource map and collect the count of each
+	// resource at the configured interval
+	resourceMetricsMu.RLock()
+	defer resourceMetricsMu.RUnlock()
+	ticker := time.NewTicker(t.duration)
+	defer ticker.Stop()
+	for metricName, resource := range resourceMetrics {
+		select {
+		case <-t.ctx.Done():
+			return t.ctx.Err()
+		case <-ticker.C:
+		}
+		count, err := t.store.GetConfigStore().Count(t.ctx, storev2.NewResourceRequestFromV2Resource(resource))
```

Comment:
This will be a bit of an issue for the "special" data types unfortunately, since they need to come from their respective store.